doc/dev/release-checklist.txt

```diff
@@ -134,6 +134,45 @@ Getting Started
 
 #. Make sure any release notes are merged into doc/release-notes/.
 
+
+How to make a final release
```

Comment:
Also rename the release notes to be '.md' to enable GitHub rich text diffs and in-browser editing